students/eowyn/session09/oo-mailroom/oomailroom.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env python
+import sys
+
+
+class Donor():
+    # Keep track of individual donors and their history of donations
+
+    def __init__(self, name):
+        self.donations = []
+        self.name = ''.join(name)  # should we check input?
+
+    def add_donation(self, amt):
+        self.donations.append(float(amt))
+
+    def generate_letter(self):
+        '''
+        Generate a formatted thank you note to a donor and return
+        the string of the thank you note for printing or writing.
+        '''
+        fs = "Thank you, {0}, for your generosity and recent gift of ${1:.2f}."
+        return fs.format(self.name, self.most_recent_donation)
+
+    @property
+    def sum_donations(self):
+        return round(sum(self.donations), 2)
+
+    @property
+    def count_donations(self):
+        return len(self.donations)
+
+    @property
+    def most_recent_donation(self):
+        return round(self.donations[-1], 2)
+
+    @property
+    def average_donation(self):
+        return round(self.sum_donations / self.count_donations, 2)
+
+
+class Transactions():
+    # Collect list of Donor objects and generate reports
+
+    def __init__(self):
+        self.all_donors = []
+
+    def add_donor(self, name, amt):
```

Comment:
this is slightly odd logic -- or a not-very-good name.

add_donor should really add a new donor.....